### DISCUSSION TITLE
<source> tags are not identified and bundled if they're used under <video> tags

### DISCUSSION THREAD

**Discussion by maxweisel** [parcel-bundler/parcel#4777] [General] [answered]
# 🐛 bug report

If I have the following html:

```
<video playsinline autoplay muted loop id="videoHeader">
  <source src="/assets/test.mp4" type="video/mp4" media="screen and (min-device-width:600px)">
  <source src="/assets/test-iphone.mp4" type="video/mp4" media="screen and (max-device-width:600px)">  -->
</video>
```

Parcel will not detect the assets in the <source> tags. It looks like this bug was previously reported and solved here: https://github.com/parcel-bundler/parcel/issues/987 However, it was only solved for `<img>` tags, and not `<video>` tags.

## 🎛 Configuration (.babelrc, package.json, cli command)

I'm not specifying a babel config.

## 🤔 Expected Behavior

Parcel should detect the source files in `<source>` tags and bundle them appropriately.

## 😯 Current Behavior

The tag is left untouched and the generated html file has a path that references an asset not in the bundle.

## 💁 Possible Solution

Ideally I'd love to see the fix for `<img>` tags applied to `<video>` tags, but I'm unsure how to do this.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.12.4
| Node             | 10.19.0
| npm/Yarn         | npm 6.14.2
| Operating System | macOS 10.15.5

**Comment by DeMoorJasper** [comment, marked as answer]
Does this work with Parcel 2?

Should work afaik: https://github.com/parcel-bundler/parcel/blob/v2/packages/transformers/html/src/dependencies.js#L14 You should probably also use relative paths

**Comment by maxweisel** [comment]
This works in parcel2, but oddly now parcel maps these to absolute paths? so it becomes `src="/test.fbab28da.mp4"`. Is there a way in parcel 2 to set it to be a relative include? I don't plan hosting the site in /, and it may actually be served from a few different places (it's a documentation site)

Max

**Comment by DeMoorJasper** [comment]
@maxweisel I'm not sure if this trick still works but you could try with `--public-url ./`

**Comment by maxweisel** [comment]
No dice, but it seems any value I pass to --public-url is ignored. I'm currently running `npx parcel build --public-url /docs ./src/index.html` and the asset path always comes back with `/test.fbab28da.mp4` in the generated html

**Reply by DeMoorJasper** [reply]
Maybe try removing `.parcel-cache` and re-running? Although it should work without doing that. I'm fairly sure the `--public-url /docs ` should work as I've used it a lot and we have integration tests for it

**Reply by maxweisel** [reply]
Hell yeah! Clearing the cache worked and the --public-url ./ trick worked flawlessly. Thank you so much for the help!

**Reply by DeMoorJasper** [reply]
Great to hear, I'll log an issue for the cache bug though, as it should work flawlessly without any need for removing the directory ever.

The issue to track the bug: https://github.com/parcel-bundler/parcel/issues/4784
